source/socket_channel_handler.c

```diff
@@ -165,9 +165,9 @@ static void s_on_readable_notification(struct aws_socket *socket, int error_code
     (void)socket;
 
     struct socket_handler *socket_handler = user_data;
-    if (!error_code) {
-        s_do_read(socket_handler);
-    } else if (!socket_handler->shutdown_in_progress) {
+    s_do_read(socket_handler);
```

Comment:
Probably should drop a little comment here to explain why we're reading when we may know that the socket is in an error state